Does not build with ghc 8.8

Discussed with my friend @guibou who is a bit more handy than me with haskell. He said he'll do a PR

```
117   │ Building library for daemons-0.2.1..
118   │ [1 of 5] Compiling Control.Pipe.Serialize ( src/Control/Pipe/Serialize.hs, dist/dist-sandbox-3b586ae8/build/Control/Pipe/Serialize.o )
119   │
120   │ src/Control/Pipe/Serialize.hs:33:16: error:
121   │     • Could not deduce (MonadFail m) arising from a use of ‘loop’
122   │       from the context: (Serialize a, Monad m)
123   │         bound by the type signature for:
124   │                    deserializer :: forall a (m :: * -> *).
125   │                                    (Serialize a, Monad m) =>
126   │                                    Pipe ByteString a m ()
127   │         at src/Control/Pipe/Serialize.hs:32:1-64
128   │       Possible fix:
129   │         add (MonadFail m) to the context of
130   │           the type signature for:
131   │             deserializer :: forall a (m :: * -> *).
132   │                             (Serialize a, Monad m) =>
133   │                             Pipe ByteString a m ()
134   │     • In the expression: loop Nothing Nothing
135   │       In an equation for ‘deserializer’:
136   │           deserializer
137   │             = loop Nothing Nothing
138   │             where
139   │                 loop mk mbin
140   │                   = do bin <- maybe await return mbin
141   │                        ....
142   │    |
143   │ 33 | deserializer = loop Nothing Nothing
144   │    |                ^^^^^^^^^^^^^^^^^^^^
```

--- a/src/Control/Pipe/Serialize.hs
+++ b/src/Control/Pipe/Serialize.hs
@@ -26,10 +26,11 @@ import Data.Serialize ( Serialize, get, encode
                       , Result(..), runGetPartial )
 import Pipes ( Pipe, await, yield )
 import Control.Monad ( forever )
+import Control.Monad.Fail ( MonadFail )
 
 -- | De-serialize data from strict 'ByteString's.  Uses @cereal@'s
 -- incremental 'Data.Serialize.Get' parser.
-deserializer :: (Serialize a, Monad m) => Pipe ByteString a m ()
+deserializer :: (Serialize a, Monad m, MonadFail m) => Pipe ByteString a m ()
 deserializer = loop Nothing Nothing
   where
     loop mk mbin = do

--- a/src/Control/Pipe/Socket.hs
+++ b/src/Control/Pipe/Socket.hs
@@ -70,7 +70,7 @@ runSocketServer lsocket handler = liftIO $ forever $ do
     (socket, _addr) <- NS.accept lsocket
     _ <- forkIO $ CE.finally
                       (handler (socketReader socket) (socketWriter socket))
-                      (NS.sClose socket)
+                      (NS.close socket)
     return ()
 
 -- | Run 'Handler' on the given socket.

--- a/src/System/Daemon.hs
+++ b/src/System/Daemon.hs
@@ -34,9 +34,9 @@ import Data.Serialize ( Serialize )
 import Data.String ( IsString(..) )
 import Network.Socket ( Socket, SockAddr(..), Family(..), SocketType(..)
                       , SocketOption(..), setSocketOption
-                      , socket, sClose, connect, bindSocket, listen
+                      , socket, close, connect, bind, listen
                       , AddrInfo(..), getAddrInfo, addrAddress, defaultHints
-                      , defaultProtocol, iNADDR_ANY, maxListenQueue )
+                      , defaultProtocol, tupleToHostAddress, maxListenQueue )
 import System.Directory ( getHomeDirectory )
 import System.FilePath ( (</>), (<.>) )
 import System.Posix.Daemon ( runDetached, isRunning )
@@ -125,7 +125,7 @@ runInForeground :: Port -> Handler () -> IO ()
 runInForeground port handler = do
     CE.bracket
         (bindPort port)
-        sClose
+        close
         (\lsocket ->
              runSocketServer lsocket handler)
 
@@ -159,7 +159,7 @@ runClientWithHandler :: HostName   -- ^ hostname
 runClientWithHandler hostname port handler = do
     CE.bracket
         (getSocket hostname port)
-        sClose
+        close
         (\s -> runSocketClient s handler)
 
 -- | Create a socket and bind it to the given port.
@@ -167,11 +167,11 @@ bindPort :: Port -> IO Socket
 bindPort port = do
     CE.bracketOnError
         (socket AF_INET Stream defaultProtocol)
-        sClose
+        close
         (\s -> do
             -- FIXME See the examples at the end of Network.Socket.ByteString
             setSocketOption s ReuseAddr 1
-            bindSocket s (SockAddrInet (fromIntegral port) iNADDR_ANY)
+            bind s (SockAddrInet (fromIntegral port) (tupleToHostAddress (0, 0, 0, 0)))
             listen s maxListenQueue
             return s)
 
@@ -183,7 +183,7 @@ getSocket hostname port = do
                              (Just $ show port)
     CE.bracketOnError
         (socket AF_INET Stream defaultProtocol)
-        sClose
+        close
         (\s -> do
              connect s (addrAddress $ head addrInfos)
              return s)

